tutorials/text/text_intro.py

```diff
@@ -93,3 +96,337 @@
+
+plt.show()
+
+##############################################################################
+#  Finally, we can specify functions for the formatter using
+# `matplotlib.ticker.FuncFormatter`.
+
+
+def formatoddticks(x, pos):
+    """Format odd tick positions
+    """
+    if x % 2:
+        return '%1.2f' % x
+    else:
+        return ''
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+ax.plot(x1, y1)
+formatter = matplotlib.ticker.FuncFormatter(formatoddticks)
+locator = matplotlib.ticker.MaxNLocator(nbins=6)
+ax.xaxis.set_major_formatter(formatter)
+ax.xaxis.set_major_locator(locator)
+
+plt.show()
+
+##############################################################################
+# Dateticks
+# ~~~~~~~~~
+#
+# Matplotlib can accept `datetime.datetime` and `numpy.datetime64`
+# objects as plotting arguments.  Dates and times require special
+# formatting, which can often benefit from manual intervention. In
+# order to help, dates have spectial Locators and Formatters,
+# defined in the `matplotlib.dates` module.
+#
+# A simple exmample is as follows.  Note how we have to rotate the
+# tick labels so that they don't over-run each other.
+import datetime
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+base = datetime.datetime(2017, 1, 1, 0, 0, 1)
+time = [base + datetime.timedelta(days=x) for x in range(len(y1))]
+
+ax.plot(time, y1)
+ax.tick_params(axis='x', rotation=70)
+plt.show()
+
+##############################################################################
+# Maybe the format of the labels above is acceptable, but the choices is
+# rather idiosyncratic.  We can make the ticks fall on the start of the month
+# by very lightly modifying `matplotlib.dates.AutoDateLocator`
+import matplotlib.dates as mdates
+
+locator = mdates.AutoDateLocator(interval_multiples=True)
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+ax.xaxis.set_major_locator(locator)
+ax.plot(time, y1)
+ax.tick_params(axis='x', rotation=70)
+plt.show()
+
+##############################################################################
+# Oddly, this changes the labels!  The easiest fix is to pass a format to
+# `matplotlib.dates.DateFormatter`.  Also note that the 29th and the next month
+# are very close together.  We can fix this by using the `dates.DayLocator`
+# class, which allows us to specify a list of days of the month to use.
+# Similar formatters are listed in the `matplotlib.dates` module.
+
+locator = mdates.DayLocator(bymonthday=[1, 15])
+formatter = mdates.DateFormatter('%b %d')
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+ax.xaxis.set_major_locator(locator)
+ax.xaxis.set_major_formatter(formatter)
+ax.plot(time, y1)
+ax.tick_params(axis='x', rotation=70)
+plt.show()
+
+##############################################################################
+# Legends and Annotations
+# =======================
+#
+#   - Legends: :ref:`sphx_glr_tutorials_intermediate_legend_guide.py`
```

Comment:
I tend to prefer
```
:doc:`/tutorials/intermediate/legend_guide`
```
which links to the page itself rather than an html anchor, and is also shorter to type.